contracts/SwitcheoTokenZIL.scilla

```diff
@@ -47,30 +55,34 @@ let get_val =
   | None => zero
   end
 
+let some_addr = 
+  fun (addr: ByStr20) =>
+  Some {ByStr20} addr
+
 (***************************************************)
 (*             The contract definition             *)
 (***************************************************)
 
 contract FungibleToken
 (
-  contract_owner: ByStr20,
-  name : String,
+  name: String,
   symbol: String,
   decimals: Uint32,
-  init_supply : Uint128
+  init_supply: Uint128,
+  init_minter: ByStr20
 )
 
 (* Mutable fields *)
 
 field total_supply : Uint128 = init_supply
 
-field balances: Map ByStr20 Uint128 
-  = let emp_map = Emp ByStr20 Uint128 in
-    builtin put emp_map contract_owner init_supply
+field balances: Map ByStr20 Uint128 = Emp ByStr20 Uint128
 
 field allowances: Map ByStr20 (Map ByStr20 Uint128) 
   = Emp ByStr20 (Map ByStr20 Uint128)
 
+field minter: ByStr20 = init_minter
```

Comment:
should this be lockproxy?

Reply:
> Need to handle minting at `transferFrom`?

I think no need and the code should be alright. But just double checking

Reply:
Yes no need. because no one is on behavior of proxy transferring tokens.